vision_datasets/key_value_pair/manifest.py

```diff
@@ -0,0 +1,128 @@
+from typing import Optional
+from ..common import MultiImageLabelManifest, MultiImageDatasetManifest, DatasetTypes
+        
+
+class KeyValuePairFieldSchema:
+    TYPE_NAME_TO_PYTHON_TYPE = {
+        'array': list,
+        'object': dict,
+        'number': (float, int),
+        'integer': int,
+        'boolean': bool,
+        'string': str, 
+        'boundingBox': list  # TODO: add more specific types such as list[int] | list[float] and fix check_field_schema_match
+    }
+    
+    def __init__(self, type: str,
+                 description: str = None,
+                 examples: list[str] = None,
+                 enum: list[str] = None,
+                 items: 'KeyValuePairFieldSchema' = None,
+                 properties: dict[str, 'KeyValuePairFieldSchema'] = None) -> None:
+        
+        self.type = type
+        self.description = description
+        self.examples = examples
+        self.enum = enum
+        self.items = KeyValuePairFieldSchema(**items) if items else None
+        self.properties = {k: KeyValuePairFieldSchema(**v) for k, v in properties.items()} if properties else None
+        self._check()
+    
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, KeyValuePairFieldSchema):
+            return False
+        return (self.type == other.type \
+            and self.description == other.description \
+            and self.examples == other.examples \
+            and self.enum == other.enum \
+            and self.items == other.items \
+            and self.properties == other.properties)
+    
+    def _check(self):
+        if self.type not in self.TYPE_NAME_TO_PYTHON_TYPE:
+            raise ValueError(f'Invalid type: {self.type}')
+        if self.enum and self.type not in ['string', 'number', 'integer']:
```

Comment:
is this case sensitive? can we check uniqueness?

Reply:
I'm not sure I understand. Do you mean checking if there are duplicates in self.enum?

Reply:
Added check for item uniqueness of enum list

Reply:
yeah, below are a few examples:
[enum1, EnUm1, Enum1]
[enum1, enum1, enum2]

Reply:
thanks! yes, it is case sensitive.